Automattic/woocommerce-payments
# Enable the use of saved SEPA payments

## Work In Progress

Fixes https://github.com/Automattic/woocommerce-payments/issues/1471

#### Changes proposed in this Pull Request
- Introduced a SEPA token class to "save SEPA payment details". Details are stored in 
- Introduced a `Payment_Method` class for `card`, `sepa_debit`. Planning on adding `sofort` and `giropay` later.
- Customers can now save SEPA payment information by clicking this checkbox: 
![image](https://user-images.githubusercontent.com/572862/114590093-8ef48c80-9c45-11eb-8cd4-7a99c843e453.png)
- Customers can now use saved SEPA payments to checkout: 
![image](https://user-images.githubusercontent.com/572862/114590442-03c7c680-9c46-11eb-86f6-a9c888c22a47.png)

#### Concerns
- SEPA token class name overlaps existing plugin's.

#### Testing instructions
Similar to the setup in https://github.com/Automattic/woocommerce-payments/pull/1537
1. Set store currency to euro
2. Enable feature flag wp option, update `_wcpay_feature_sepa` to `1`
3. Enable SEPA in checkbox at bottom of `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments`
4. Add item to cart and go to checkout
5. You should see this screen, type in SEPA debit card number, click "Save payment...". Checkout 
![image](https://user-images.githubusercontent.com/572862/114595858-1513d180-9c4c-11eb-90d6-a340ce628fa9.png)
6. Checkout should be successful. Purchase another item. Now select the SEPA payment we just saved, 
![image](https://user-images.githubusercontent.com/572862/114596020-50ae9b80-9c4c-11eb-92d6-9609f65c11cf.png)
7. The checkout should proceed successfully with the saved SEPA card.



-------------------
